timur/lib/client/jsx/utils/query_filter_path_builder.ts

```diff
@@ -29,34 +24,27 @@ export default class QueryFilterPathBuilder {
     let nestedFilterIndex: number = 0;
 
     this.path.forEach((modelName: string) => {
-      if (stepIsOneToMany(this.models, previousModelName, modelName)) {
```

Comment:
I suggest viewing this in github with `?w=1` so the whitespace diff of tabbing is dropped.
Essentially I drop the stepIsOneToMany conditional, then I pick the folding predicate clause based on entry in anyMap (essentially, non one to many relationships only have one valid predicate here).

The previous approach just appends the model (out of order) at the end, and not following the injection logic.